File: deltav/src/main/java/com/ithit/webdav/samples/deltavservlet/websocket/NotificationEncoder.java
package com.ithit.webdav.samples.deltavservlet.websocket;

import javax.websocket.EncodeException;
import javax.websocket.Encoder;
import javax.websocket.EndpointConfig;

/**
 * Encodes notification object to the JSON
 */
public class NotificationEncoder implements Encoder.Text<WebSocketServer.Notification> {

    @Override
    public void init(EndpointConfig config) {
    }

    @Override
    public void destroy() {
    }

    @Override
    public String encode(WebSocketServer.Notification notification) throws EncodeException {
        return "{" +
                "\"folderPath\" : \"" + notification.getFolderPath() + "\" ," +
                "\"eventType\" : \"" + notification.getEventType() + "\"" +
                "}";
    }
}

File: Kotlin/kotlinfs/src/main/kotlin/com/ithit/webdav/samples/kotlinfs/websocket/NotificationEncoder.kt
package com.ithit.webdav.samples.kotlinfs.websocket

import javax.websocket.EncodeException
import javax.websocket.Encoder
import javax.websocket.EndpointConfig

/**
 * Encodes notification object to the JSON
 */
class NotificationEncoder : Encoder.Text<WebSocketServer.Notification> {

    override fun init(config: EndpointConfig) {}

    override fun destroy() {}

    @Throws(EncodeException::class)
    override fun encode(notification: WebSocketServer.Notification): String {
        return "{" +
                "\"folderPath\" : \"" + notification.folderPath + "\" ," +
                "\"eventType\" : \"" + notification.eventType + "\"" +
                "}"
    }
}
